modules/collect/src/main/java/com/opengamma/strata/collect/id/StandardId.java

```diff
@@ -117,6 +122,18 @@ public static StandardId parse(String str) {
     return new StandardId(str.substring(0, pos), str.substring(pos + 1));
   }
 
+  /**
+   * Encode a string suitable for use as the scheme.
+   * <p>
+   * This uses percent encoding, just like URI.
+   * 
+   * @param scheme  the scheme to encode
+   * @return the encoded scheme
+   */
+  public static String encodeScheme(String scheme) {
+    return SCHEME_ESCAPER.escape(scheme);
```

Comment:
There's probably an implication that the result is a valid scheme, but the scheme regex mandates one letter to begin with, which it looks like this wouldn't enforce.